src/router/index.js

```diff
@@ -1,24 +1,35 @@
-import { parse, stringify } from "query-string";
+const storeAndQuery = rawQuery => {
+  const params = decodeURIComponent(rawQuery)
+    .replace(/^\?/, "")
+    .split("&");
+
+  const query = params.filter(pair => !pair.startsWith("storeData=")).join("&");
+  const storeDatas = params.filter(pair => pair.startsWith("storeData="));
```

Comment:
A very minor optimisation, but you could collapse that into a single `reduce()`:

```js
return params.reduce(
  (acc, pair) => {
    if (!acc.storeData && pair.startsWith('storeData='))
      acc.storeData = decodeURIComponent(pair.replace(/^storeData=/, ''));
    else
      acc.query += (acc.query ? '&' : '') + pair;
    return acc;
  },
  { query: '', storeData: null }
);
```

Reply:
Additionally, I'm thinking that it might be better to take the latest `storeData` found in the URL instead of taking the first one. Not quite sure what the convention here should be, though. In that case, you could remove the `!acc.storeData` condition.